app/react/Settings/components/LanguageList.tsx

```diff
@@ -0,0 +1,243 @@
+              <ResetDefaultTranslationButton
+                onClick={() => {
+                  setResettingLanguage(language);
+                }}
+                className={!language.translationAvailable ? 'action-hidden' : ''}
+              />
+            </div>
+            <div className="col">
+              <SetAsDefaultButton
+                className={language.default ? 'btn-success' : ''}
+                onClick={!language.default ? () => setDefaultLanguage(language.key) : () => {}}
+              />
+            </div>
+            <div className="col">
+              <DeleteButton
+                onClick={() => setDeletingLanguage(language)}
+                className={language.default ? 'action-hidden' : ''}
+              />
+            </div>
+          </div>
+        ))}
+      </div>
+      <div className="panel-heading">
+        <Translate>Available Languages</Translate>
+      </div>
+      <div className="available-languages">
+        {availableLanguages.map(language => (
+          <div key={language.key} className="row">
+            <div className="col">
+              {`${language.label} (${language.key}) `}
+              {language.translationAvailable && <TranslationAvailable />}
+            </div>
+            <div className="col">
+              <button
+                type="button"
+                onClick={() => {
+                  setAddingLanguage(language);
+                }}
+                className="btn btn-success btn-xs template-remove"
+              >
+                <Icon icon="plus" /> &nbsp;
+                <span>
+                  <Translate>Add language</Translate>
+                </span>
+              </button>
+            </div>
+          </div>
+        ))}
+      </div>
+
+      {addingLanguage && (
+        <Confirm
+          accept={() => {
+            addLanguage({ ...addingLanguage });
+            setAddingLanguage(undefined);
+          }}
+          cancel={() => {
+            setAddingLanguage(undefined);
+          }}
+          title={
+            <>
+              <Translate>Confirm add</Translate>&nbsp;{addingLanguage.label}
+            </>
+          }
+          message="This action may take some time while we add the extra language to the entire collection."
+          extraConfirm
+          type="success"
+        />
+      )}
+      {deletingLanguage && (
+        <Confirm
+          accept={() => {
+            deleteLanguage(deletingLanguage.key);
+            setDeletingLanguage(undefined);
+          }}
+          cancel={() => {
+            setDeletingLanguage(undefined);
+          }}
+          title={
+            <>
+              <Translate>Confirm delete </Translate>&nbsp;{deletingLanguage.label}
+            </>
+          }
+          message={
+            <>
+              <Translate>Are you sure you want to delete</Translate>&nbsp; {deletingLanguage.label}
+              <Translate> language?</Translate>
```

Comment:
The space before and after `language?`would be better handled with `&nbsp;`

Reply:
absolutely